internal/sources/cloudmonitoring/cloud_monitoring.go

```diff
@@ -29,6 +29,26 @@ import (
 
 const SourceKind string = "cloud-monitoring"
 
+type userAgentRoundTripper struct {
+	userAgent string
+	next      http.RoundTripper
+}
+
+func (rt *userAgentRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+	newReq := *req
+	newReq.Header = make(http.Header)
+	for k, v := range req.Header {
+		newReq.Header[k] = v
+	}
+	ua := newReq.Header.Get("User-Agent")
+	if ua == "" {
+		newReq.Header.Set("User-Agent", rt.userAgent)
+	} else {
+		newReq.Header.Set("User-Agent", rt.userAgent+" "+ua)
```

Comment:
Correct me if I'm wrong but seems like best practice when appending UA is to append to the back instead of the front? 

```suggestion
		newReq.Header.Set("User-Agent", ua+" "+rt.userAgent)
```

Reply:
Following the same format as done in csqladmin source. 

https://github.com/googleapis/genai-toolbox/blob/main/internal/sources/cloudsqladmin/cloud_sql_admin.go#L38C1-L51C2

Reply:
Made the change as suggested after discussing with @prernakakkar-google